crates/tako/src/server/task.rs

```diff
@@ -58,18 +59,53 @@ bitflags::bitflags! {
     }
 }
 
+pub struct TaskInput {
+    task: TaskRef,
+    output_id: u32, // MAX = pure dependency on task, not real output id
```

Comment:
Let's use an enum for this, instead of some magic MAX value. We can use `NonZero` for the output ID to avoid increasing the size of the enum.

Reply:
The problem is that 0 is valid output_id so I need something like NonMax if I do not want some "+1/-1" magic. I know it seems 1quite a premature optimization but I expect a lot of TaskInputs.

Reply:
What about something like this? https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2f37ca86e5a006308703f8010a55ff85

Reply:
Oookay. I somehow still like my max solution, but you are right :)